### ISSUE TITLE
Log non-null field values of deleted records

### ISSUE THREAD

**Issue by timo11** [specify/specify7#594]
see #498

**Comment by timo11** [comment]
values of a deleted record's nonnull fields (besides 'version') are logged

**Comment by timo11** [comment]
This fix is on the summer19 and develop branches.

**Comment by maxxxxxdlp** [comment]
Logs properly, except for some reason the newest row after each delete is always an Update
![Screen Shot 2020-03-13 at 1 43 39 PM](https://user-images.githubusercontent.com/40512816/76650473-b9e74b00-6530-11ea-8350-8303ab28e966.png)

**Comment by timo11** [comment]
I am unable to duplicate this issue.

**Comment by maxxxxxdlp** [comment]
TL;DR; querying auditing table returns results in a wrong order

I tried to replicate that in Sp7 `TestBiology8`

After modifying global prefs, I created and removed a Locality
The auditing log table showed several `Add ` actions for `SpQuery` as if they happened after deleting the Locality

Then, I created and removed an Agent. These actions were logged between Locality and SpQuery, which is weird

Probably `SpQuery` logs have a wrong timestamp, making these actions look as if they were more recent, though it does not seem so when I took a look at the SQL export of the DB

<img width="1439" alt="Screen Shot 2020-03-17 at 9 38 06 AM" src="https://user-images.githubusercontent.com/40512816/76867101-139e8c80-6833-11ea-86f0-8f97d9e442cb.png">
<img width="1440" alt="Screen Shot 2020-03-17 at 9 37 51 AM" src="https://user-images.githubusercontent.com/40512816/76867104-139e8c80-6833-11ea-8cd2-56350adb55fe.png">

**Comment by timo11** [comment]
I'm not sure anything is wrong. Aren't you adding and modifying a query to query the audit log?

@benanhalt  -- It is not happening for me, but the first case for this bug with the updates made me think 7 might be 'pre-saving' the locality so the user's save becomes an update. Does 7 ever do that i?

**Comment by maxxxxxdlp** [comment]
@timo11 SpQuery logging is still acting weird
You can test it in `TestBiology10` but I am pretty sure it happens for every database
After creating a new database, there will be 32 auditing records in it (for Discipline, Institution and etc)
Specify 7 also logs the changes to the saved queries; Specify 6 does not do that
The changes to the query are logged properly, but they get sorted weirdly (tested when sorted by date DESC)
For some reason, new SpQuery records are shown at the beginning and at the end of the auditing table, making it appear if some of them were modified even before the database was created, which is not true
If it was to work properly, all of the query records should have been shown at the beginning (because of the DESC sorting which shows newer first)

![Screenshot from 2020-03-17 14-25-57](https://user-images.githubusercontent.com/40512816/76893974-3f356d00-685b-11ea-860e-bc9692589112.png)

When I exported the database from SP7, the dates on the records seems to be in the proper order (but you can still check it), so it is likely an issue with sorting

**Comment by timo11** [comment]
I haven't seen any thing out of order in the mysql tables. I think Sp7 is sorting on the date rather than the complete timestamp.

**Comment by benanhalt** [comment]
@maxxxxxdlp can you create a separate issue for the date vs datetime thing and reference it here? Then this issue can be resolved because it doesn't have anything to do with the original issue, right?
